buildSrc/src/main/kotlin/ktlint-kotlin-common.gradle.kts

```diff
@@ -0,0 +1,32 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    id("org.jetbrains.kotlin.jvm")
+}
+
+repositories {
+    mavenCentral()
+}
+
+kotlin {
+    jvmToolchain {
+        (this as JavaToolchainSpec).languageVersion.set(JavaLanguageVersion.of(8))
```

Comment:
Out of curiocity, as I'm not sure I get toolchains properly: Wouldn't you want to explicitly set target version? I can imagine at some point you would like to bump toolchain version to Java 17, but the output artifact should remain compatible with Java 8. 
If someone changes the toolchain version it will affect produced artifacts which is an unwanted behavior, right? 

Basically I understood you always want to compile the project with highest possible (java and kotlin) versions, but java target version and kotlin api version should be set to the lower library supports - is that wrong assumption?

Reply:
> Wouldn't you want to explicitly set target version?

Toolchain by default sets target version to be equal to the toolchain one if it is not set explicitly.

>  If someone changes the toolchain version it will affect produced artifacts which is an unwanted behavior, right?

Right, but such change should go via PR review :wink:

> Basically I understood you always want to compile the project with highest possible (java and kotlin) versions, but java target version and kotlin api version should be set to the lower library supports

Initially I want to use JDK 1.8 to compile and add later tests for CLI that runs against latest JDK version. Generally it is also possible to use toolchain with JDK 17 + set target to "1.8". That change could be done at a later time.

Reply:
Great, that was the answer I was looking for. Thanks for explanation 👍